### PR TITLE
Docs: groups ppl criterion

### PR DESCRIPTION
**Pull request by ZPain8464**
Fixes https://github.com/pomerium/documentation/issues/1146.

### CODE REVIEW

File: content/docs/capabilities/ppl.mdx [hunk 1]
```diff
@@ -132,24 +136,22 @@ PPL supports many different criteria:
...
 | `cors_preflight` | Anything. Typically `true`. | Returns true if the incoming request uses the `OPTIONS` method and has both the `Access-Control-Request-Method` and `Origin` headers. Used to allow [CORS pre-flight requests]. |
+| \* `date` | [Date Matcher] | Returns true if the time of the request matches the constraints. |
+| \* `day_of_week` | [Day of Week Matcher] | Returns true if the day of the request matches the constraints. |
 | `device` | [Device matcher] | Returns true if the incoming request includes a valid device ID or type. |
 | `domain` | [String Matcher] | Returns true if the logged-in user's email address domain (the part after `@`) matches the given value. |
 | `email` | [String Matcher] | Returns true if the logged-in user's email address matches the given value. |
+| \* `groups` | [String Matcher] | Returns true if a user's group ID matches the supplied value **exactly**. `groups` data is only available after a successful directory sync. See [Identity Providers](/docs/identity-providers) for vendor-specific directory sync steps. |
```
Comment (wasaga) [inline content/docs/capabilities/ppl.mdx:144]: I believe `groups` is not using String Matcher, but rather `String List Matcher` that has a single operator `has` that checks that a given string is present in a given list of strings.

**Review by wasaga — CHANGES_REQUESTED** [review]

File: content/docs/capabilities/ppl.mdx [hunk 2]
```diff
@@ -304,6 +306,25 @@ allow:
         starts_with: 'admin@'
 ```
 
+### String List Matcher
+
+A string list matcher is an object that supports a single `has` operator as a key. The `has` operator checks that a given string is present in a list of strings.
+
+The `groups` and `record` criteria both support the `has` operator.
```
Comment (wasaga) [inline content/docs/capabilities/ppl.mdx:313]: the example below does not demonstrate `has` operator for `record`

Comment (ZPain8464) [reply content/docs/capabilities/ppl.mdx:313]: @wasaga I meant to add that. Thanks for catching it. I've updated the example.

File: content/docs/capabilities/ppl.mdx [hunk 3]
```diff
@@ -322,7 +322,7 @@ allow:
     - record:
         type: pomerium.io/ExternalDataSource
         field: id
```
Comment (wasaga) [inline content/docs/capabilities/ppl.mdx:324]: `id` field is not a list 

I believe `has` is only applicable to `foreign_field`. 

@calebdoxsey do you have a good example maybe for `has` usage in `record` ?

Comment (calebdoxsey) [reply content/docs/capabilities/ppl.mdx:324]:
```yaml
allow:
  and:
  - record:
      type: example.com/geoip
      field: country
      is: "US"
  - record:
      type: example.com/hr_user
      field: departments
      has: "engineering"
```

Comment (ZPain8464) [reply content/docs/capabilities/ppl.mdx:324]: @calebdoxsey  Is this part necessary?  Or is this just showing another operator that also uses list string matching?
```yaml
  - record:
      type: example.com/geoip
      field: country
      is: "US"
```

Comment (calebdoxsey) [reply content/docs/capabilities/ppl.mdx:324]: Right that's showing using `is`. The record operator supports `contains`, `ends_with`, `is`, `starts_with` and `has`. It depends on what data is stored. If it's a list you'd want to use `has`. If it's a string you'd want to use one of the string matchers.

**Review by wasaga — APPROVED** [review]

**Pull request merged**